Assets/UI/PartialScreens/citystates_CQUI.lua

```diff
@@ -0,0 +1,106 @@
+-- ===========================================================================
+-- Cached Base Functions
+-- ===========================================================================
+BASE_AddCityStateRow = AddCityStateRow;
+
+-- ===========================================================================
+--  CONSTANTS
+-- ===========================================================================
+local COLOR_ICON_BONUS_OFF:number = UI.GetColorValueFromHexLiteral(0xff606060);
+
+-- ===========================================================================
+--  CQUI Function Extensions
+-- ===========================================================================
+function AddCityStateRow( kCityState:table )
+    local kInst = BASE_AddCityStateRow(kCityState);
+
+    local envoyTable:table = {};
```

Comment:
The older version had this line
`
kInst.Suzerain:SetText( CQUI_TruncateSuzerainName(kCityState.SuzerainName) );
`
I don't see this in the extension function, was the removal intentional?

Reply:
not intentional, just missed it because it was above the "-- Begin CQUI Changes Marker" on line 921 and I didn't notice the diff when comparing in notepad++.

Reply:
good catch, too
without that truncation:
![image](https://user-images.githubusercontent.com/8787640/88464418-6cc89d80-ce6f-11ea-97df-3d316d7f2e56.png)

with the truncation:
![image](https://user-images.githubusercontent.com/8787640/88464424-7c47e680-ce6f-11ea-9429-96d7459abfd6.png)

Reply:
I was just about to comment that maybe the truncation functions are not needed. I was playing a game and noticed that for me the suzerain was already truncated (I'm playing with GatheringStorm turned on, I'm guessing you are not).
I was actually thinking a better way to do the truncation is via the XML if you look at "\Assets\Expansion2\Replacements\citystates.xml" you'll see that we have `TruncateWidth="70"` set for the Suzerain.

I would like to propose getting rid of the truncate function and instead rely on the XML and set `TruncateWidth="60"` for Suzerain and SecondHighestName. The one in function relies on counting the number of characters, but the font here is not a fixed width font, so counting letters is not the ideal way to do truncation. The XML is based on whatever they have for screen units, I'm proposing 60 since in my game the tag for Poundmaker overlaps with the number of envoys when it goes to double digits.
Using XML truncation only
![image](https://user-images.githubusercontent.com/178540/88464775-6fd18700-ce8b-11ea-8a27-9f643ea0272f.png)
Using the github version
![image](https://user-images.githubusercontent.com/178540/88464860-1584f600-ce8c-11ea-8eed-e47fde3e365c.png)

Reply:
i will try it out